src/NzbDrone.Core/Notifications/Webhook/WebhookService.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Specialized;
+using System.IO;
+using System.Linq;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Processes;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.Tv;
+using NzbDrone.Core.Validation;
+using NzbDrone.Core.Rest;
+using RestSharp;
+using NzbDrone.Common.Exceptron.fastJSON;
+
+namespace NzbDrone.Core.Notifications.Webhook
+{
+    public interface IWebhookService
+    {
+        void OnDownload(Series series, EpisodeFile episodeFile, WebhookSettings settings);
+        void OnRename(Series series, WebhookSettings settings);
+        void OnGrab(string grabMessage, WebhookSettings settings);
+        ValidationFailure Test(WebhookSettings settings);
+    }
+
+    public class WebhookService : IWebhookService
+    {
+        private readonly IProcessProvider _processProvider;
+        private readonly Logger _logger;
+
+        public WebhookService(IProcessProvider processProvider, Logger logger)
+        {
+            _processProvider = processProvider;
+            _logger = logger;
+        }
+
+        public void OnDownload(Series series, EpisodeFile episodeFile, WebhookSettings settings)
+        {
+            WebhookPayload payload = new WebhookPayload
+            {
+                EventType = "Download",
+                Series = new WebhookSeries(series),
+                EpisodeFile = new WebhookEpisodeFile(series, episodeFile)
+            };
+
+            NotifyWebhook(payload, settings);
+        }
+
+        public void OnRename(Series series, WebhookSettings settings)
+        {
+            WebhookPayload payload = new WebhookPayload
+            {
+                EventType = "Rename",
+                Series = new WebhookSeries(series)
+            };
+
+            NotifyWebhook(payload, settings);
+        }
+        
+        public void OnGrab(string grabMessage, WebhookSettings settings)
+        {
+            NotifyWebhook(new { EventType = "Grab", Message = grabMessage }, settings);
+        }
+
+        public void NotifyWebhook(object body, WebhookSettings settings)
+        {
+            try {
+                var client = RestClientFactory.BuildClient(settings.Url);
+                var request = new RestRequest(Method.POST);
+                request.RequestFormat = DataFormat.Json;
```

Comment:
I'm thinking we might want to make the `ContentType` and `Method` configurable as part of settings, but not something we need right now,